persistence/aggregate.go

```diff
@@ -32,6 +35,21 @@ type AggregateMetaData struct {
 	BarrierEventID string
 }
 
+type AggregateSnapshot struct {
+	// State is the instance's current aggregate state.
+	State dogma.AggregateRoot
+
+	// HandlerKey is the identity key of the aggregate message handler.
+	HandlerKey string
+
+	// InstanceID is the aggregate instance ID.
+	InstanceID string
+
+	// Revision is the instance's current version, used to enforce optimistic
+	// concurrency control.
+	Revision uint64
```

Comment:
```suggestion
	// Revision is the instance's current version.
	Revision uint64
```

We don't need to enforce OCC when persisting a snapshot because it doesn't matter if it's out of date - we'll just replay any events that happened since on top of it. This has some positive implications, like the fact that we don't _have_ to persist it in the same transaction as other aggregate options.